### DISCUSSION TITLE
Production use cases

### DISCUSSION THREAD

**Discussion by ghost** [uNetworking/uWebSockets#1081] [Show and tell]
We know that former v0.14 is in use pretty much all over the internet in all kinds of shapes and forms - from the most popular bitcoin sites to the most popular "io-games" and in various widespread "support chatting" services and I can't possibly keep track of it all (and I wouldn't want to).

So what about the new v18 codebase? We've survived the shitstorm that followed v0.14 deprecation, the forks of it, and today there's definitely a substantial interest in this new code. Not only does v18 perform 25% better than v0.14, it has more high-level features that companies need to make great apps (such as pub/sub).

As a showcase of what companies have built, and a point of reference for newcomers, I'm listing a few entires I know use the latest v18 code in production. I did include Emupedia even if it's not a commercial product - it's too cool to not list.

Bitfinex being the most ambitious and obvious poster-boy for now; it trades 100 million USD every 24 hours, via µWS v18.

<table>
<tr>
<td>Bitfinex</td>
<td>Godmodetrader</td>
<td>Guidants</td>
</tr>
<tr>
<td><img src="https://user-images.githubusercontent.com/3397140/87223833-87374d00-c380-11ea-9231-e77e5572029c.png" height="300"></td>
<td><img src="https://user-images.githubusercontent.com/3397140/87223830-830b2f80-c380-11ea-8a45-a675111b97e1.png" height="300"></td>
<td><img src="https://user-images.githubusercontent.com/3397140/87223832-856d8980-c380-11ea-8227-539e9bf75fe9.png" height="300"></td>
</tr>
<tr>
<td>Emupedia</td>
</tr>
<tr>
<td><img src="https://user-images.githubusercontent.com/3397140/87224165-2a896180-c383-11ea-9436-6dd467166b40.png" height="300"></td>
</tr>
</table>

If you know any company use of µWS v18, or use it in production yourself, feel free to post it here. It has to be running in some kind of production, by an actual company - I do not want open source projects listed here. I also do not want v0.14 use cases listed here.

**Comment by ghost** [comment]
These run v0.17 in production

Zlap.io
![zlap](https://user-images.githubusercontent.com/3397140/87564287-8116d880-c6c0-11ea-89d6-b56ff1dd1825.png)
Parabol.co
![parabol](https://user-images.githubusercontent.com/3397140/87564480-bcb1a280-c6c0-11ea-83fd-68571a9413c6.png)

**Comment by hermanbanken** [comment]
https://ws.meethue.com (Philips Hue) is clusterWS currently (after 0.14 stuff), but we're [updating to 18/19](https://ws-dev.meethue.com/).
Still seeing some issues that are blocking our release though.

**Reply by ghost** [reply]
Kool

**Comment by morajabi** [comment]
We've migrated to uWebSockets and love it @ https://usenoor.com
